File: api/app/Models/Estudiante.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Estudiante extends Model
{
    use HasFactory;

    protected $fillable = [
        'codigo_rude',
        'cedula_identidad',
        'nombre',
        'apellido_paterno',
        'apellido_materno',
        'genero',
        'fecha_nacimiento'
    ];

    public $timestamps = false;

    // One To Many -Una Licencia puede tener uno o muchos estudiantes
    public function licencias()
    {
      return $this->hasMany(Licencia::class);
    }

    public function asistencias()
    {
      return $this->hasMany(Asistencia::class);
    }

    //RELACION MUCHOS A MUCHOS (ENTRE CURSO-ESTUDIATE)
    public function cursos(){
      return $this->belongsToMany('App\Models\Curso');
    }

    //RELACION MUCHOS A MUCHOS (ENTRE ESTUDIATE-MATERIA)
    public function materias(){
      return $this->belongsToMany(Materia::class,'estudiante_materia');
    }

}

File: apk/lib/Models/estudiante.dart

class Estudiante {
    Estudiante({
      required  this.id,
      required  this.codigoRude,
      required  this.cedulaIdentidad,
      required  this.nombre,
      required  this.apellidoPaterno,
      required  this.apellidoMaterno,
      required  this.genero,
      required  this.fechaNacimiento,
    });

    int id;
    String codigoRude;
    String cedulaIdentidad;
    String nombre;
    String apellidoPaterno;
    String apellidoMaterno;
    String genero;
    String fechaNacimiento;

    factory Estudiante.fromJson(Map<String, dynamic> json) => Estudiante(
        id: json["id"],
        codigoRude: json["codigo_rude"],
        cedulaIdentidad: json["cedula_identidad"],
        nombre: json["nombre"],
        apellidoPaterno: json["apellido_paterno"],
        apellidoMaterno: json["apellido_materno"],
        genero: json["genero"],
        fechaNacimiento: json["fecha_nacimiento"].toString(),
    );

    Map<String, dynamic> toJson() => {
        "id": id,
        "codigo_rude": codigoRude,
        "cedula_identidad": cedulaIdentidad,
        "nombre": nombre,
        "apellido_paterno": apellidoPaterno,
        "apellido_materno": apellidoMaterno,
        "genero": genero,
        "fecha_nacimiento": fechaNacimiento,
    };
}